flink-tests/src/test/java/org/apache/flink/test/javaApiOperators/ExecutionEnvironmentITCase.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.test.javaApiOperators;
+
+import org.apache.flink.api.java.DataSet;
+import org.apache.flink.api.java.ExecutionEnvironment;
+import org.apache.flink.configuration.ConfigConstants;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.test.javaApiOperators.util.CollectionDataSets;
+import org.apache.flink.test.util.MultipleProgramsTestBase;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+
+@RunWith(Parameterized.class)
+public class ExecutionEnvironmentITCase extends MultipleProgramsTestBase {
+
+
+	public ExecutionEnvironmentITCase(ExecutionMode mode) {
+		super(mode);
+	}
+
+	@Parameterized.Parameters(name = "Execution mode = {0}")
+	public static Collection<ExecutionMode[]> executionModes(){
+		Collection<ExecutionMode[]> c = new ArrayList<ExecutionMode[]>(1);
+		c.add(new ExecutionMode[] {ExecutionMode.CLUSTER});
+		return c;
+	}
+
+
+	@Test
+	public void testLocalEnvironmentWithConfig() throws Exception {
+		IllegalArgumentException e = null;
+		try {
+			Configuration conf = new Configuration();
+			conf.setBoolean(ConfigConstants.FILESYSTEM_DEFAULT_OVERWRITE_KEY, true);
+			conf.setString(ConfigConstants.TASK_MANAGER_TMP_DIR_KEY, "/tmp/thelikelyhoodthatthisdirectoryexisitsisreallylow");
+			conf.setInteger(ConfigConstants.AKKA_ASK_TIMEOUT, 3);
+			final ExecutionEnvironment env = ExecutionEnvironment.createLocalEnvironment(conf);
+
+			DataSet<Integer> ds = CollectionDataSets.getIntegerDataSet(env);
+			ds.writeAsText(null);
+			env.execute();
+		} catch(IllegalArgumentException iae) {
```

Comment:
Why does this cause an `IllegalArgumentException? This seems like the wrong exception type ;-)